db/knex_migrations/2024-04-30-0000-tls.js

```diff
@@ -0,0 +1,24 @@
+exports.up = function (knex) {
+    return knex.schema
+        .alterTable("monitor", function (table) {
+            table.text("tcp_request").defaultTo(null);
+            table.boolean("tcp_enable_tls").notNullable().defaultTo(false);
```

Comment:
Probably I'm missing something but I can't find a "regular tls boolean field". The only fields containing `tls` in the name are:

> grpc_enable_tls (boolean)
> ignore_tls (boolean)
> tls_ca (text)
> tls_cert (text)
> tls_key (text)

`ignore_tls` could conceivably be abused for this but it stands for "Ignore TLS/SSL errors for HTTPS websites", so that would seem a bit of a stretch.

Reply:
I see where you are coming from. The grpc field should not exist.

In an ideal world, both (tcp+grpc) of these monitoes would use the same mechanism as http:
- `expiryNotifcation` controls if tls expiry is a notification
- `ignoreTls` controls if tls errors are notifications 

Being transparent:
The crosstalk (expiry is currently an tls error 😅) between the two options is a bit unfortunate and not clear to our users.

Reply:
I plan to eventually add support for `expiry_notification` and `ignore_tls` (which should really be called `ignore_tls_errors`) because that has been requested in the original PR as well. But I might do that as a separate PR because there's always "just one more feature" that could be added to any PR. :-)

The two options do have a very different function, though, so if it's not clear to users, that seems like a documentation issue. Sometimes a bit of explanatory text in the UI can go a long way.

As for the "enable TLS" setting, what do you suggest as the way forward? Is renaming `grpc_enable_tls` to `enable_tls` a possibility as part of the DB migration? If so, at least GRPC and TCP could use the same setting. Otherwise that would leave us with the generic name `enable_tls` (which other monitors could use down the line) or the monitor-specific `tcp_enable_tls`

Reply:
> I might do that as a separate PR

Good choice 👍🏻

> that seems like a documentation issue

Definitvely. Also the expiry buttons could be disabled if tls is ignored (different issue)

> Is renaming `grpc_enable_tls` to `enable_tls` a possibility as part of the DB migration?

I think this should be merged into `ignore_tls(_errors)` instead.
(I agree on the renaming, but lets keep this PR focused on one thing..)

I don't think a separate `enable_tls` needs to exist, right?

Reply:
(`NOT ignoreTls` seems equivalent to `grpc_enable_tls`)

Reply:
With the TLS monitor as it is right now, `enable_tls == true` is implied. But once I merge it with the TCP monitor, we'll definitely need to have a setting that determines how to connect:

1. `enable_tls == false` ⇒ unencrypted connection (= TCP monitor today)
2. `enable_tls == true && tcp_start_tls == false` ⇒ Native TLS connection
3. `enable_tls == true && tcp_start_tls == true` ⇒ Start off unencrypted, TLS after STARTTLS handshake

Once implemented (separate PR), the two existing settings would become relevant for the merged TCP/TLS monitor as well:

`ignore_tls[_errors]`: Don't fail on TLS errors like "wrong host", "self-signed certificate", etc.
`expiry_notification`: Warn if the certificate is going to expire soon.